gpytorch/utils/masked_linear_operator.py

```diff
@@ -0,0 +1,96 @@
+from typing import Optional, Union
```

Comment:
Small nit: could this move to `gpytorch/lazy/`, and could you add a small unit test for it (the unit test should be simple if you use the `linear_operator/test/linear_operator_test_case` classes.)

Reply:
Ah, I just saw https://github.com/cornellius-gp/linear_operator/pull/69 - let's maybe wait to merge in this PR until the MaskedLinearOperator PR is in!

Reply:
So I guess we now wait for a linear_operator release?